jerry-core/parser/js/js-lexer.c

```diff
@@ -2291,7 +2289,9 @@ lexer_expect_object_literal_id (parser_context_t *context_p, /**< context */
         lexer_skip_spaces (context_p);
 
         if (context_p->source_p < context_p->source_end_p
-            && context_p->source_p[0] != LIT_CHAR_COLON)
+            && context_p->source_p[0] != LIT_CHAR_COLON
+            && context_p->source_p[0] != LIT_CHAR_RIGHT_BRACE
+            && context_p->source_p[0] != LIT_CHAR_COMMA)
```

Comment:
What about the pre scanner?

Reply:
Sorry I'm not sure I understand this comment.

Reply:
I mean the pre-scanner which checks JS code early. It is used by `switch`, `for-in`, etc.